### ISSUE TITLE
Is there a way to customize truncated dir?

### ISSUE THREAD

**Issue by peacock0803sz** [romkatv/powerlevel10k#1124]
Thank you for your theme that is fast and configurable.

## tl;dr

I want to customize truncate dir via editing `~/.p10k.zsh` or another way.
E.g. `~/.ghq/github.com/peaocck0803sz/some_repo` -> `WORK/some_repo`

## My environment

```
$ zsh --version
zsh 5.8 (x86_64-apple-darwin18.7.0)
```

## Details

My configs around dir in `~/.p10k.zsh` is here, but it doesn't work I expected.
Default truncations work certainly in a narrow terminal window.

![image](https://user-images.githubusercontent.com/33555487/100130771-808c1b00-2ec6-11eb-971e-f855e7bab85d.png)

```zsh
  typeset -g POWERLEVEL9K_SHORTEN_STRATEGY=truncate_to_unique
  typeset -g POWERLEVEL9K_SHORTEN_DELIMITER=
  local anchor_files=(  # not changed from default
  )
  typeset -g POWERLEVEL9K_SHORTEN_FOLDER_MARKER="(${(j:|:)anchor_files})"
  typeset -g POWERLEVEL9K_DIR_TRUNCATE_BEFORE_MARKER=false
  typeset -g POWERLEVEL9K_SHORTEN_DIR_LENGTH=1
  typeset -g POWERLEVEL9K_DIR_SHOW_WRITABLE=v2

  typeset -g POWERLEVEL9K_DIR_WORK_VISUAL_IDENTIFIER_EXPANSION=''
  typeset -g POWERLEVEL9K_DIR_WORK_FOREGROUND=39
  typeset -g POWERLEVEL9K_DIR_WORK_SHORTENED_FOREGROUND=103
  typeset -g POWERLEVEL9K_DIR_WORK_ANCHOR_FOREGROUND=39

  typeset -g POWERLEVEL9K_DIR_WORK_NOT_WRITABLE_VISUAL_IDENTIFIER_EXPANSION='!'
  typeset -g POWERLEVEL9K_DIR_WORK_NOT_WRITABLE_FOREGROUND=red
  typeset -g POWERLEVEL9K_DIR_WORK_NOT_WRITABLE_SHORTENED_FOREGROUND=red
  typeset -g POWERLEVEL9K_DIR_WORK_NOT_WRITABLE_ANCHOR_FOREGROUND=red

  typeset -g POWERLEVEL9K_DIR_CLASSES=(
    "$GHQ_WORKDIR(|/*)" WORK 'WORK'
  )
```

And `export GHQ_WORKDIR="~/.ghq/github.com/peacock0803sz"` in `~/.zshenv`

My dotfile's all is [here](https://github.com/peacock0803sz/dotfiles/).

Thank you.

**Comment by romkatv** [comment]
Add this to `~/.zshrc`:

```zsh
hash -d work=~/.ghq/github.com/peaocck0803sz/some_repo
```

This will establish a bidirectional mapping between `~work` and `~/.ghq/github.com/peaocck0803sz/some_repo ` which works just like the mapping between `~` and `$HOME`. You can type `cd ~work/phone.app.locales` and it'll just work. Tab completion will work, too (`cd ~work/<TAB>`). Prompt will look nicer, etc.

**Comment by peacock0803sz** [comment]
It works correctly but looks like magic to me 😇 

![image](https://user-images.githubusercontent.com/33555487/100135424-f5faea00-2ecc-11eb-9184-6559451e853c.png)

Thanks kindly and speedy response!

**Comment by romkatv** [comment]
This is a zsh feature. It’s not specific to powerlevel10k. Super useful though!
